projects/openapi-utilities/src/diff/__tests__/diff.test.ts

```diff
@@ -128,11 +133,562 @@ describe('diff openapi', () => {
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+        {
+          paths: {
+            '/user/{user_Id}': {
+              get: {
+                responses: {},
+                requestBody: {
+                  content: {
+                    'application/json': {
+                      schema: {
+                        type: 'object',
+                        properties: {},
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        }
+      );
+      expect(diffNoChanges.length).toBe(0);
+
+      expect(diffWithChanges.length).toBe(1);
+      expect(diffWithChanges).toMatchSnapshot();
+    });
   });
 
   test('diff with empty array', () => {
     expect(diff({}, after)).toMatchSnapshot();
   });
+
+  describe('object <> allOf compares properties properly', () => {
+    test('no diff on refactor to allOf', () => {
+      const objectBefore: FlatOpenAPIV3.SchemaObject = {
+        type: 'object',
+        properties: {
+          a: { type: 'string' },
+          b: { type: 'number' },
+        },
+      };
+
+      const allOfAfter: FlatOpenAPIV3.SchemaObject = {
+        allOf: [
+          {
+            type: 'object',
+            properties: {
+              b: { type: 'number' },
+            },
+          },
+          {
+            type: 'object',
+            properties: {
+              a: { type: 'string' },
+            },
+          },
+        ],
+      };
+
+      const diffResult = diff(
+        objectBefore,
+        allOfAfter,
+        jsonPointerHelpers.compile([
+          'paths',
+          '/users',
+          'get',
+          'responses',
+          '200',
+          'content',
+          'application/json',
+          'schema',
+        ])
+      );
+
+      expect(diffResult.length).toBe(0);
+    });
+    test('property diff is correct', () => {
+      const objectBefore: FlatOpenAPIV3.SchemaObject = {
+        type: 'object',
+        properties: {
+          a: { type: 'string' },
+          b: { type: 'number' },
+        },
+      };
+
+      const allOfAfter: FlatOpenAPIV3.SchemaObject = {
+        allOf: [
+          {
+            type: 'object',
+            properties: {
+              b: { type: 'number' },
+            },
+          },
+          {
+            type: 'object',
+            required: ['a'],
+            properties: {
+              a: { type: 'number' },
+            },
+          },
+        ],
+      };
+
+      const diffResult = diff(
+        objectBefore,
+        allOfAfter,
+        jsonPointerHelpers.compile([
+          'paths',
+          '/users',
+          'get',
+          'responses',
+          '200',
+          'content',
+          'application/json',
+          'schema',
+        ])
+      );
+
+      expect(diffResult.length).toBe(2);
+    });
+    test('removals of properties', () => {
```

Comment:
This is a case where we won't be able to show anything in changelog (if you look at the trail generated `"before": "/paths/~1users/get/responses/200/content/application~1json/schema/required/0",` you can see that it will point to something that does not exist in the after case.

Reply:
I may be missing something but how is this any different than removing a normal property? We have to anchor that somewhere (its parent)? 

We also have this issue in the normal changelog for the required if you go from `required: ['a']` -> `_` instead of `required: []`. 

The before paths don't need to resolve in the after spec, that's why there are separate paths right?